### DISCUSSION TITLE
Digital Signatures and text boxes would be very helpful features

### DISCUSSION THREAD

**Discussion by dlebedinsky** [sumatrapdfreader/sumatrapdf#4554] [Ideas]
Right now, Adobe PDF lets you add your signature to a PDF, as a paid feature. It's incredibly annoying that they paywalled it, because it is a useful feature. Whenever I have to sign a government or work-related document, I find myself having to print it, fill it out and add my signature, scan it, and email it back to them, which is a total waste of time. If sumatra pdf let users draw their signature with a mouse and insert it to the document without having to print it, and add text boxes to the pdf (like if the form needs you to enter your address or phone number, for example), that would be great. 

TLDR; I just want to be able to fill out and sign forms without having to print them or pay for proprietary software.

**Comment by GitHubRulesOK** [comment]
adding an image clip is not impossible and has been done in SumatraPDF by others but it not a simple addition as it should be done Adobes way ! I think there is a test case here https://github.com/sumatrapdfreader/sumatrapdf/discussions/3489 technically such objects would show as Adobe formatted "stamps"
![image](https://github.com/user-attachments/assets/d5ae64d3-2932-463a-aacd-7bb11fab15b4)

**Comment by mngerhold-crypto** [comment]
I would also like to be able to 'sign' pdfs: but with the existing annotation feature, one can, say, highlight a selection, and then in the annotation editor add something to **contents**, eg 'signed by MNG 8/9/25'.  If this is saved to the (or another) pdf, then the words will show as a popup when the annotation is hovered over:
<img width="428" height="199" alt="signed" src="https://github.com/user-attachments/assets/d7cb37ad-0a4e-43eb-8224-8c8af3e0bf1e" />
